app/src/org/commcare/gis/MapboxLocationPickerActivity.kt

```diff
@@ -0,0 +1,277 @@
+        if (!inViewMode()) {
+            map.addOnMapClickListener(mapClickListener)
+        }
+        setMapStyle()
+    }
+
+    private fun setMapStyle() {
+        val style = Style.Builder()
+                .fromUri(mapStyles[currentMapStyleIndex])
+                .withImage(MARKER_ICON_IMAGE_ID, ContextCompat.getDrawable(this, R.drawable.marker)!!)
+        map.setStyle(style) {
+            enableLocationComponent(it)
+            addMarker(it, map.cameraPosition.target)
+            initialMarkerPosition()
+        }
+    }
+
+    private fun enableLocationComponent(style: Style) {
+        val component = map.locationComponent
+        component.activateLocationComponent(LocationComponentActivationOptions.builder(
+                this, style
+        ).build())
+        component.isLocationComponentEnabled = true
+        component.cameraMode = CameraMode.TRACKING
+        component.renderMode = RenderMode.NORMAL
+    }
+
+    private fun addMarker(style: Style, point: LatLng) {
+        symbolManager = SymbolManager(mapView, map, style)
+        symbolManager?.let { symbolManager ->
+            symbolManager.iconAllowOverlap = true
+            val symbolOptions = SymbolOptions()
+                    .withLatLng(point)
+                    .withIconImage(MARKER_ICON_IMAGE_ID)
+            symbol = symbolManager.create(symbolOptions)
+        }
+    }
+
+    private fun initialMarkerPosition() {
+        if (cameraPosition != null) {
+            cameraPosition?.let { pos ->
+                viewModel.reverseGeocode(pos.target)
+                updateMarker(pos.target)
+            }
+        } else {
+            map.locationComponent.locationEngine
+                    ?.getLastLocation(object : LocationEngineCallback<LocationEngineResult> {
+                        override fun onSuccess(result: LocationEngineResult?) {
+                            result ?: return
+                            result.lastLocation?.let { location ->
+                                val point = LatLng(location.latitude, location.longitude, location.altitude)
+                                viewModel.reverseGeocode(point)
+                                updateMarker(point)
+                            }
+                        }
+
+                        override fun onFailure(exception: Exception) {}
+                    })
+        }
+    }
+
+    private fun updateMarker(point: LatLng) {
+        symbol?.let { symbol ->
+            symbol.latLng = point
+            symbolManager?.update(symbol)
+            val pos = CameraPosition.Builder()
+                    .target(point)
+                    .zoom(15.0)
+                    .build()
+            map.animateCamera(CameraUpdateFactory.newCameraPosition(pos), 10)
+        }
+    }
+
+    override fun onLocationRequestStart() {
+        /// Do nothing
+    }
+
+    override fun onLocationResult(result: Location) {
+        if (isManualSelectedLocation) {
```

Comment:
Any reason it doesn't have all the checks from [here](https://github.com/dimagi/commcare-android/blob/master/app/src/org/commcare/activities/GeoPointMapActivity.java#L165-L169) ?

Reply:
Ohh Thanks for pointing both these.

Reply:
there are still other checks like the one with accuracy thats not here.